corehq/apps/sms/tests/test_phone_number_in_multiple_domains.py

```diff
@@ -0,0 +1,181 @@
+        "required": 0,
+        "ix": "1",
+        "help": None,
+        "output": "<xml>dummy form xml</xml>"
+    }
+}
+
+
+@flag_enabled("ONE_PHONE_NUMBER_MULTIPLE_CONTACTS")
+@flag_enabled("INBOUND_SMS_LENIENCY")
+@patch('corehq.apps.smsforms.util.critical_section_for_smsforms_sessions',
+       new=lambda contact_id: contextlib.nullcontext())
+class FormSessionMultipleContactsTestCase(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.domain_name1 = "domain1"
+        cls.domain_name2 = "domain2"
+
+        cls.backend = SQLTestSMSBackend.objects.create(
+            name='BACKEND',
+            domain=cls.domain_name1,
+            is_global=False,
+            hq_api_id=SQLTestSMSBackend.get_api_id()
+        )
+
+        # share the backend with the 2nd domain
+        MobileBackendInvitation.objects.create(domain=cls.domain_name2, backend=cls.backend, accepted=True)
+
+        # set up the backend mapping
+        SQLMobileBackendMapping.set_default_domain_backend(cls.domain_name1, cls.backend)
+        SQLMobileBackendMapping.set_default_domain_backend(cls.domain_name2, cls.backend)
+
+        cls.phone_number = '01112223333'
+        cls.number1 = PhoneNumber.objects.create(
+            domain=cls.domain_name1,
+            owner_doc_type='CommCareCase',
+            owner_id='fake-owner-id1',
+            phone_number=cls.phone_number,
+            backend_id=None,
+            ivr_backend_id=None,
+            verified=True,
+            is_two_way=True,
+            pending_verification=False,
+            contact_last_modified=datetime.utcnow()
+        )
+
+        cls.number2 = PhoneNumber.objects.create(
+            domain=cls.domain_name2,
+            owner_doc_type='CommCareCase',
+            owner_id='fake-owner-id2',
+            phone_number=cls.phone_number,
+            backend_id=None,
+            ivr_backend_id=None,
+            verified=True,
+            is_two_way=False,
+            pending_verification=False,
+            contact_last_modified=datetime.utcnow()
+        )
+
+    def setUp(self):
+        get_channel_for_contact.clear(self.number1.owner_id, self.number1.phone_number)
+        get_channel_for_contact.clear(self.number2.owner_id, self.number2.phone_number)
+
+    @patch('corehq.apps.sms.api.get_location_id_by_verified_number', MagicMock(return_value=None))
+    @patch('corehq.apps.sms.api._domain_accepts_inbound', MagicMock(return_value=True))
+    @patch('corehq.apps.sms.api.domain_has_privilege', MagicMock(return_value=True))
+    @patch('corehq.apps.sms.api._allow_load_handlers', MagicMock(return_value=True))
+    @patch('corehq.apps.smsforms.util.submit_form_locally', MagicMock(
+        return_value=Mock(xform=Mock(form_id="123"))))
+    def test_sms_form_session_across_multiple_domains_with_same_backend_and_number(self):
+        self._test(self.number1)  # test session from domain1
+        self._test(self.number2)  # test session from domain2 immediately after
```

Comment:
@Robert-Costello I don't actually think it's useful to have the two tests right after each-other. I think we could restructure this a into 4 separate test cases:

@Robert-Costello I think it would be useful to have the following test cases:

* session workflow with number1 with the '+' prefix on the SMS number
* session workflow with number1 without the '+' prefix on the SMS number
* session workflow with number2 with the '+' prefix on the SMS number
* session workflow with number2 without the '+' prefix on the SMS number